test/cpp/jit/test_gpu.cpp

```diff
@@ -18148,6 +18148,135 @@ TEST(NVFuserTest, FusionRfactorContigIDs_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
 }
 
+TEST(NVFuserTest, FusionMMASwizzlePrimitiveFloatAcc_CUDA) {
+  FusionExecutor fe;
+
+  std::string kernel = R"(
+  __global__ void mma32x32x8_mini_gemm_float(Tensor<__half, 2> Ag, Tensor<__half, 2> Bg, Tensor<float, 2> Cg0, Tensor<float, 2> Cg1){
+    // Accumulator
+    float C[32];
+    // initialize C:
+    for(int i=0;i<32;i++){
+        C[i] = 0;
+    }
+    // Allocate Smem
+    __shared__ uint4 As_mem[32*8 / 8];
+    __shared__ uint4 Bs_mem[8*32 / 8];
+    auto As = reinterpret_cast<__half*>(As_mem);
```

Comment:
Just curious here, does it make any difference if it's cast to `__shared__ half*`? I remember I've read somewhere that nvcc would use a generic load ptx instruction as opposed to a shmem load instruction if a pointer is not annotated as shared. This is just a test function, so it doesn't matter at all, but I'm just curious if we should generally be aware of pointer types .

Reply:
Thanks for pointing this out. 

It looks like nvcc does some inference here in this case. But there's no public docs on how powerful these state space inference can go. 
```
        st.shared.u32 	[%rd66], %r46;
	st.shared.u32 	[%rd66+4], %r47;
	st.shared.u32 	[%rd66+8], %r48;
	st.shared.u32 	[%rd66+12], %r49;
	...
	ld.shared.v4.u32 	{%r66, %r67, %r68, %r69}, [%rd75];
```

I guess it wouldn't hurt to add them. We could consider cleaning these up in follow ups since we do something similar for other shared mem allocs too. [https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/codegen.cpp#L1276-L1279](https://github.com/csarofeen/pytorch/blob/devel/torch/csrc/jit/codegen/cuda/codegen.cpp#L1276-L1279)